simd/src/Kokkos_SIMD.hpp

```diff
@@ -84,33 +95,45 @@ namespace simd_abi {
 namespace Impl {
 
 #if defined(KOKKOS_ARCH_AVX512XEON)
+template <class T>
 using host_fixed_native = avx512_fixed_size<8>;
 template <int N>
 using host_native_abi = avx512_fixed_size<N>;
 
 #elif defined(KOKKOS_ARCH_AVX2)
+template <class T>
 using host_fixed_native = avx2_fixed_size<4>;
 template <int N>
 using host_native_abi = avx2_fixed_size<N>;
 
+#elif defined(KOKKOS_ARCH_ARM_SVE)
+template <class T>
+using host_fixed_native =
+    sve_fixed_size<(__ARM_FEATURE_SVE_BITS / (8 * sizeof(T)))>;
```

Comment:
I am pretty sure our previous use of these host_fixed_native thingies was so that a SIMD of double and SIMD of float etc. have all the same size (number of elements). I.e. that you could in principle multiple simd of double and simd of float or at least assign them to each other. In that spirit I think the template needs to go and `host_fixed_native = see_fixed_size<(__ARM_FEATURE_SVE_BITS/sizeof(double))>`

@ldh4 is my understanding correct?